### PR TITLE
ByteArrayToStringConverter

### PR DESCRIPTION
**Pull request by pm48**
Addition of byte Array to String Converter and its test.

### CODE REVIEW

File: library/src/main/java/com/datatorrent/lib/converter/ByteArrayToStringConverterOperator.java [hunk 1]
```diff
@@ -20,15 +20,13 @@
...
 public class ByteArrayToStringConverterOperator extends BaseOperator
 {
   private Charset characterEncoding;
 
-  public ByteArrayToStringConverterOperator()
-  {
-    characterEncoding = Charset.forName("UTF-8");
-  }
-
   public Charset getCharacterEncoding()
```
Comment (siyuanh) [inline library/src/main/java/com/datatorrent/lib/converter/ByteArrayToStringConverterOperator.java]: Please have string type getter/setter method(you can convert in the getter/setter method), so the App Builder can handle

File: library/src/test/java/com/datatorrent/lib/converter/ByteArrayToStringConverterTest.java [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+    TestUtils.setSink(testop.output, testsink);
+    testop.beginWindow(0);
+    testop.setCharacterEncoding(Charset.forName("UTF-8"));
+    testop.input.put(utf8Bytes);
+    testop.setCharacterEncoding(Charset.forName("UTF-16"));
+    testop.input.put(asciiBytes);
+    testop.endWindow();
+
+    Assert.assertEquals(2,testsink.collectedTuples.size());
+    for (String output: testsink.collectedTuples) {
```
Comment (siyuanh) [inline library/src/test/java/com/datatorrent/lib/converter/ByteArrayToStringConverterTest.java]: This for loop is not very readable and only work for 2 elements, you can create an expected list and assertequals on expected and actual list. And also add one test for default charset

File: library/src/test/java/com/datatorrent/lib/converter/ByteArrayToStringConverterTest.java [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+import org.slf4j.LoggerFactory;
+
+public class ByteArrayToStringConverterTest
+{
+  @Test
+  public void testByteArrayToStringConversion() throws UnsupportedEncodingException
+  {
+    ByteArrayToStringConverterOperator testop = new ByteArrayToStringConverterOperator();
+    String test1 = "hello world with UTF8";
+    byte[] utf8Bytes = test1.getBytes("UTF8");
```
Comment (siyuanh) [inline library/src/test/java/com/datatorrent/lib/converter/ByteArrayToStringConverterTest.java]: typo here?

File: library/src/test/java/com/datatorrent/lib/converter/ByteArrayToStringConverterTest.java (hunk 2 above)
Comment (pm48) [inline library/src/test/java/com/datatorrent/lib/converter/ByteArrayToStringConverterTest.java]: The test for default charset is already there. Changed data type of charset to String for app builder use. Please see below commit.

**Pull request closed without merging**